src/change-location-link/change-location-link.component.tsx

```diff
@@ -0,0 +1,59 @@
+import React from "react";
+import Button from "carbon-components-react/es/components/Button";
+import { useTranslation } from "react-i18next";
+import { BrowserRouter, Redirect } from "react-router-dom";
+import styles from "./change-location.link.component.scss";
+import Location20 from "@carbon/icons-react/es/location/20";
+const openmrsSpaBase = window["getOpenmrsSpaBase"]();
+
+interface ChangeLocationLinkProps {
+  referer?: string;
+  currentLocation: string;
+}
+
+const ChangeLocationLink: React.FC<ChangeLocationLinkProps> = ({
+  referer,
+  currentLocation,
+}) => {
+  const { t } = useTranslation();
+  const [triggerLocationChange, setTriggerLocationChange] = React.useState(
+    false
+  );
+
+  const changeLocation = () => {
+    setTriggerLocationChange((prevState) => !prevState);
+  };
+
+  return (
+    <BrowserRouter>
```

Comment:
Why all this?

Just use the `navigate` function of `esm-framework`. No need for this ceremony.

Reply:
Here are [the docs for navigate](https://github.com/openmrs/openmrs-esm-core/blob/master/packages/esm-config/README.md#navigate) @donaldkibet . They'll probably go someplace that makes more sense soon, as we refactor to use esm-framework.

Reply:
> Why all this?
> 
> Just use the `navigate` function of `esm-framework`. No need for this ceremony.

the `navigate` function doesn't provide the ability to define the `Referer` attribute hence the use of the `Redirect`

Reply:
> Why all this?
> 
> Just use the `navigate` function of `esm-framework`. No need for this ceremony.

`navigate` doesn't provide the ability to define the `referer` param required to redirect the user from the `url` the change location action is triggered from.

Reply:
Then we either should add this or change the mechanism. I guess a changed mechanism would make sense here. I rather see the redirect as part of the URL.

Reply:
@FlorianRappl  I have made changes to use the `returnToUrl`